File: core/designsystem-core/src/commonMain/kotlin/com/minddistrict/multiplatformpoc/core/designsystem/core/PokemonTypeColors.kt
package com.minddistrict.multiplatformpoc.core.designsystem.core

import androidx.compose.ui.graphics.Color

/**
 * Pokémon Type Colors (Domain-Specific)
 * 
 * These colors represent the official Pokémon type system and are shared across all design systems.
 * Colors are adjusted for WCAG AA accessibility compliance (4.5:1 contrast ratio on appropriate backgrounds).
 * 
 * Use these colors for:
 * - Type badges on Pokémon cards
 * - Type indicators in detail views
 * - Color-coding for type effectiveness
 */
object PokemonTypeColors {
    val Fire = Color(0xFFFF4422)
    val Water = Color(0xFF3399FF)
    val Grass = Color(0xFF77CC55)
    val Electric = Color(0xFFFFCC33)
    val Psychic = Color(0xFFFF5599)
    val Ice = Color(0xFF66CCFF)
    val Dragon = Color(0xFF7766EE)
    val Dark = Color(0xFF775544)
    val Fairy = Color(0xFFEE99EE)
    val Normal = Color(0xFFAAAA99)
    val Fighting = Color(0xFFBB5544)
    val Flying = Color(0xFF8899FF)
    val Poison = Color(0xFF9955BB)
    val Ground = Color(0xFFDDBB55)
    val Rock = Color(0xFFBBAA77)
    val Bug = Color(0xFFAABB22)
    val Ghost = Color(0xFF6666BB)
    val Steel = Color(0xFFAAAABB)
    
    /**
     * Get background color for a Pokémon type with theme-aware adjustments.
     * @param type The Pokémon type name (e.g., "fire", "water")
     * @param isDark Whether dark mode is active
     * @return Color for the type background
     */
    fun getBackground(type: String, isDark: Boolean): Color {
        val normalizedType = type.lowercase()
        val baseColor = when (normalizedType) {
            "normal" -> Normal
            "fire" -> Fire
            "water" -> Water
            "electric" -> Electric
            "grass" -> Grass
            "ice" -> Ice
            "fighting" -> Fighting
            "poison" -> Poison
            "ground" -> Ground
            "flying" -> Flying
            "psychic" -> Psychic
            "bug" -> Bug
            "rock" -> Rock
            "ghost" -> Ghost
            "dragon" -> Dragon
            "dark" -> Dark
            "steel" -> Steel
            "fairy" -> Fairy
            else -> Normal
        }
        
        // Adjust brightness for theme
        return if (isDark) {
            // Lighten for dark mode (better contrast on dark backgrounds)
            baseColor.copy(
                red = (baseColor.red * 1.2f).coerceIn(0f, 1f),
                green = (baseColor.green * 1.2f).coerceIn(0f, 1f),
                blue = (baseColor.blue * 1.2f).coerceIn(0f, 1f)
            )
        } else {
            // Slightly darken for light mode (better contrast on light backgrounds)
            baseColor.copy(
                red = (baseColor.red * 0.9f).coerceIn(0f, 1f),
                green = (baseColor.green * 0.9f).coerceIn(0f, 1f),
                blue = (baseColor.blue * 0.9f).coerceIn(0f, 1f)
            )
        }
    }
    
    /**
     * Get content (text) color that contrasts well with the type background.
     * @param type The Pokémon type name
     * @param isDark Whether dark mode is active
     * @return Color for text on the type background
     */
    fun getContent(type: String, isDark: Boolean): Color {
        // Most type colors work well with white text
        return Color.White
    }
    
    /**
     * Get both background and content colors as a Pair.
     */
    fun getColors(type: String, isDark: Boolean): Pair<Color, Color> {
        return Pair(getBackground(type, isDark), getContent(type, isDark))
    }
}

File: iosApp/iosApp/DesignSystem/PokemonTypeColors.swift
import SwiftUI

/**
 * Pokémon Type Colors (Domain-Specific)
 * 
 * Matches PokemonTypeColors from Compose Multiplatform.
 * Colors are adjusted for WCAG AA accessibility (4.5:1 contrast).
 */
enum PokemonType: String, CaseIterable {
    case normal, fire, water, electric, grass, ice, fighting, poison
    case ground, flying, psychic, bug, rock, ghost, dragon, dark, steel, fairy
    
    var color: Color {
        switch self {
        case .fire: return Color(red: 1.0, green: 0.267, blue: 0.133)     // #FF4422
        case .water: return Color(red: 0.2, green: 0.6, blue: 1.0)        // #3399FF
        case .grass: return Color(red: 0.467, green: 0.8, blue: 0.333)    // #77CC55
        case .electric: return Color(red: 1.0, green: 0.8, blue: 0.2)     // #FFCC33
        case .psychic: return Color(red: 1.0, green: 0.333, blue: 0.6)    // #FF5599
        case .ice: return Color(red: 0.4, green: 0.8, blue: 1.0)          // #66CCFF
        case .dragon: return Color(red: 0.467, green: 0.4, blue: 0.933)   // #7766EE
        case .dark: return Color(red: 0.467, green: 0.333, blue: 0.267)   // #775544
        case .fairy: return Color(red: 0.933, green: 0.6, blue: 0.933)    // #EE99EE
        case .normal: return Color(red: 0.667, green: 0.667, blue: 0.6)   // #AAAA99
        case .fighting: return Color(red: 0.733, green: 0.333, blue: 0.267)  // #BB5544
        case .flying: return Color(red: 0.533, green: 0.6, blue: 1.0)     // #8899FF
        case .poison: return Color(red: 0.6, green: 0.333, blue: 0.733)   // #9955BB
        case .ground: return Color(red: 0.867, green: 0.733, blue: 0.333) // #DDBB55
        case .rock: return Color(red: 0.733, green: 0.667, blue: 0.467)   // #BBAA77
        case .bug: return Color(red: 0.667, green: 0.733, blue: 0.133)    // #AABB22
        case .ghost: return Color(red: 0.4, green: 0.4, blue: 0.733)      // #6666BB
        case .steel: return Color(red: 0.667, green: 0.667, blue: 0.733)  // #AAAABB
        }
    }
    
    /// Get background color with theme-aware adjustments
    func backgroundColor(isDark: Bool) -> Color {
        if isDark {
            // Lighten for dark mode (better contrast)
            return color.opacity(0.8)
        } else {
            // Slightly darken for light mode
            return color
        }
    }
    
    /// Get content (text) color that contrasts with background
    var contentColor: Color {
        // Light types use dark text
        switch self {
        case .electric, .ice, .fairy, .normal:
            return .black
        default:
            return .white
        }
    }
    
    /// Initialize from string (case-insensitive)
    init?(string: String) {
        guard let type = PokemonType(rawValue: string.lowercased()) else {
            return nil
        }
        self = type
    }
}
